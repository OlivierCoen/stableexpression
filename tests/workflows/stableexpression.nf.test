nextflow_workflow {

    name "Test Workflow STABLEEXPRESSION"
    script "workflows/stableexpression.nf"
    workflow "STABLEEXPRESSION"
    tag "workflow"

    test("Two Expression Atlas accessions provided") {

        tag "workflow_eatlas_accessions"

        when {
            params {
                outdir = "results"
                species = 'solanum tuberosum'
                eatlas_accessions = "E-MTAB-552,E-GEOD-61690"
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }
    }

    test("Two Expression Atlas no keyword (whole species)") {

        tag "workflow_eatlas_no_kw"

        when {
            params {
                outdir = "results"
                species = 'solanum tuberosum'
                fetch_eatlas_accessions = true
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }
    }

    test("Two Expression Atlas keywords provided") {

        tag "workflow_eatlas_kw"

        when {
            params {
                outdir = "results"
                species = 'solanum tuberosum'
                eatlas_keywords = "potato,stress"
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }
    }

    test("Two Expression Atlas accessions provided - normalisation with EdgeR") {

        tag "workflow_eatlas_accessions"

        when {
            params {
                outdir = "results"
                species = 'solanum tuberosum'
                eatlas_accessions = "E-MTAB-552,E-GEOD-61690"
                normalisation_method = "edger"
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }
    }

    test("Two Expression Atlas accessions provided - skip no zero filter") {

        tag "workflow_eatlas_accessions"

        when {
            params {
                outdir = "results"
                species = 'solanum tuberosum'
                eatlas_accessions = "E-MTAB-552,E-GEOD-61690"
                allow_zeros = true
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }
    }

    test("Full workflow - Expression Atlas only") {

        when {
            params {
                outdir = "results"
                species = 'solanum tuberosum'
                eatlas_accessions = "E-MTAB-552,E-GEOD-61690"
                eatlas_keywords = "phloem"
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
