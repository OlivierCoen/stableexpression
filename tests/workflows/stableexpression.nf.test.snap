{
    "Full workflow - Expression Atlas only": {
        "content": [
            {
                "0": [
                    "variation_coefficients.csv:md5,2c9749853675f7359ebf2cd0242641fe"
                ],
                "ch_output_from_variation_coefficient": [
                    "variation_coefficients.csv:md5,2c9749853675f7359ebf2cd0242641fe"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-31T21:10:44.253721"
    },
    "Two Expression Atlas accessions provided - skip no zero normalisation filter": {
        "content": [
            {
                "0": [
                    "variation_coefficients.csv:md5,baaa5074dcc665399f74d80a2e48bb89"
                ],
                "ch_output_from_variation_coefficient": [
                    "variation_coefficients.csv:md5,baaa5074dcc665399f74d80a2e48bb89"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-31T13:44:57.610154"
    },
    "Two Expression Atlas keywords provided - normalisation with EdgeR": {
        "content": [
            {
                "0": [
                    "variation_coefficients.csv:md5,cd96cd99f15072771afa6180092c7bc2"
                ],
                "ch_output_from_variation_coefficient": [
                    "variation_coefficients.csv:md5,cd96cd99f15072771afa6180092c7bc2"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-30T18:54:55.39010153"
    },
    "Two Expression Atlas accessions provided - skip no zero filter": {
        "content": [
            {
                "0": [
                    "variation_coefficients.csv:md5,9ef07615dab2a2242076220054dbc1ff"
                ],
                "ch_output_from_variation_coefficient": [
                    "variation_coefficients.csv:md5,9ef07615dab2a2242076220054dbc1ff"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-31T21:09:26.917847"
    },
    "Two Expression Atlas keywords provided": {
        "content": [
            {
                "0": [
                    "variation_coefficients.csv:md5,c17c7c45d599dcd1611e2bbd4264fce7"
                ],
                "ch_output_from_variation_coefficient": [
                    "variation_coefficients.csv:md5,c17c7c45d599dcd1611e2bbd4264fce7"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-31T21:07:48.666413"
    },
    "Two Expression Atlas no keyword (whole species)": {
        "content": [
            {
                "0": [
                    "variation_coefficients.csv:md5,354aa7ee34a20ebf4b09d3be78c54194"
                ],
                "ch_output_from_variation_coefficient": [
                    "variation_coefficients.csv:md5,354aa7ee34a20ebf4b09d3be78c54194"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-03T11:21:34.40984841"
    },
    "Two Expression Atlas accessions provided - normalisation with EdgeR": {
        "content": [
            {
                "0": [
                    "variation_coefficients.csv:md5,6fe277e27eb841547c0e76da1d808a4e"
                ],
                "ch_output_from_variation_coefficient": [
                    "variation_coefficients.csv:md5,6fe277e27eb841547c0e76da1d808a4e"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-31T21:08:35.886211"
    },
    "Two Expression Atlas accessions provided": {
        "content": [
            {
                "0": [
                    "variation_coefficients.csv:md5,9ef07615dab2a2242076220054dbc1ff"
                ],
                "ch_output_from_variation_coefficient": [
                    "variation_coefficients.csv:md5,9ef07615dab2a2242076220054dbc1ff"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2024-12-31T21:03:43.75912"
    }
}